search/includes/classes/class-queue.php

```diff
@@ -1080,18 +1080,25 @@ public function get_average_queue_wait_time() {
 
 		$table_name = $this->schema->get_table_name();
 
-		$average_wait_time = $wpdb->get_var(
+		$queue_stats = $wpdb->get_row(
+			// Cannot prepare table name. @codingStandardsIgnoreStart
 			$wpdb->prepare(
-				"SELECT FLOOR( AVG( TIMESTAMPDIFF( SECOND, queued_time, NOW() ) ) ) AS average_wait_time FROM $table_name WHERE 1" // Cannot prepare table name. @codingStandardsIgnoreLine
+				"SELECT
+					FLOOR( AVG( TIMESTAMPDIFF( SECOND, queued_time, NOW() ) ) ) AS average_wait_time,
+					FLOOR( MAX( TIMESTAMPDIFF( SECOND, queued_time, NOW() ) ) ) AS longest_wait_time,
+					COUNT( * ) AS queue_count
+				FROM $table_name
+				WHERE 1"
 			)
+			// @codingStandardsIgnoreEnd
 		);
 
 		// Null value will usually mean empty table
-		if ( is_null( $average_wait_time ) || ! is_numeric( $average_wait_time ) ) {
+		if ( is_null( $queue_stats ) || ! is_object( $queue_stats ) ) {
 			return 0;
```

Comment:
Maybe we could return `null` or `false`, now that the happy path returns an object.

Reply:
We should make sure the return signature is consistent, with this PR `get_queue_stats` now can return either an integer 0 or an object of `average_wait_time, longest_wait_time, queue_count` which seems a bit explode-y.

In fact, I think the unhappy path will throw a warning now (same with null of course):

```
php > $b = 0;
php > intval( $b->my_unhappy_var );
PHP Warning:  Attempt to read property "my_unhappy_var" on int in php shell code on line 1
```

I think probably what makes sense from a defensive standpoint is to have those values set to 0 inside the `if` block.  I.e.
`$queue_stats = (object) [ 'average_wait_time' => 0, 'longest_wait_time' => 0, 'queue_count' => 0 ];`

Reply:
Great point, fixed!